catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/SetVariableTest.java

```diff
@@ -0,0 +1,158 @@
+public class SetVariableTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	public void createProject() {
+		SetVariableBrick setVariableTestBrick1 = new SetVariableBrick(10);
+		SetVariableBrick setVariableTestBrick2 = new SetVariableBrick(0);
+		Script script = BrickTestUtils.createProjectAndGetStartScript("setVariableTest");
+		script.addBrick(setVariableTestBrick1);
+		script.addBrick(setVariableTestBrick2);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testCreateNewUserVariableAndDeletion() {
+		String userVariableName = "testVariable1";
+		String secondUserVariableName = "testVariable2";
+
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_set_variable);
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(userVariableName);
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(secondUserVariableName);
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_set_variable_edit_text)
+				.perform(click());
+
+		deleteItemWithTextFromFormulaEditorDataList(userVariableName);
+		pressBack();
+		onView(withText(userVariableName))
+				.check(doesNotExist());
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(userVariableName))
+				.check(doesNotExist());
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(secondUserVariableName))
+				.check(matches(isDisplayed()));
+	}
+
+
+	//TODO: Move this to a Formula Editor Test, which?
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testCreateUserVariableInFormulaEditor() {
+		String userVariableName = "testVariable1";
+		addNewVariableViaFormulaEditor(1, userVariableName);
+		onView(allOf(withParent(withId(R.id.set_variable_spinner)), withText(userVariableName)))
+				.check(matches(isDisplayed()));
+		pressBack();
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner).onChildView(withText(userVariableName))
+				.check(matches(isDisplayed()));
+		onBrickAtPosition(1)
+				.checkShowsText(userVariableName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testIfSetVariableSpinnerCanHoldMultipleVariables() {
+		String userVariableName = "testVariable1";
+		String userVariableNameTwo = "testVariable2";
+		addNewVariableViaFormulaEditor(1, userVariableName);
+		onView(withId(R.id.formula_editor_keyboard_ok))
+				.perform(click());
+		onBrickAtPosition(1).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(userVariableNameTwo);
+		onBrickAtPosition(2).onVariableSpinner(R.id.set_variable_spinner)
```

Comment:
dont do this, it clicks on the spinner twice.
please write proper methods for this.

have a look at https://github.com/Catrobat/Catroid/pull/2656/files
or wait until it is merged (hopefully very soon) and rebase your PR and use his methods.